lib/irmin.mli

```diff
@@ -379,7 +379,22 @@ module type AO = sig
       responsibility of the append-only store to generate a
       consistent key. *)
 
-end
+  end
+
+(** Append-only store for links *)
+module type AO_LINK = sig
+
+
+  (** {1 Append-only Link stores} *)
+
+  include RO
+
+  val add: t -> key -> value -> key Lwt.t
```

Comment:
I think I would use a different function name, as `add` is already used with a different type somewhere else. What abouht `add_link`?